files/default/neutron-ha-tool.py

```diff
@@ -867,6 +870,41 @@ def pick(self):
         return random.choice(self.agents)
 
 
+class LeastBusyAgentPicker(object):
+    def __init__(self, qclient, agents):
+        self.now = datetime.datetime.now
+        self.cache_created_at = None
+        self.qclient = qclient
+        self.agents_by_id = {agent['id']: agent for agent in agents}
+        self.router_count_per_agent_id = {}
+        self.refresh_router_count_per_agent_id()
+
+    def refresh_router_count_per_agent_id(self):
+        LOG.info("Refreshing router count per agent cache")
+        self.router_count_per_agent_id = dict()
+        for agent_id in self.agents_by_id:
+            self.router_count_per_agent_id[agent_id] = len(
+                list_routers_on_l3_agent(self.qclient, agent_id)
+            )
+        self.cache_created_at = self.now()
+
+    def cache_expired(self):
+        cache_life = self.now() - self.cache_created_at
+        return cache_life.seconds > ROUTER_CACHE_MAX_AGE_SECONDS
```

Comment:
Right, good spot. Look at this:
```python
>>> import datetime
>>> now =datetime.datetime.now()
>>> tomorrow = now + datetime.timedelta(days=1)
>>> (tomorrow - now).seconds
0
>>> (tomorrow - now).total_seconds()
86400.0
```

I guess I'll use `total_seconds` as I did initially <- @aplanas